### PR TITLE
Expand vitessce to full window, with snackbar

### PR DESCRIPTION
**Pull request by keller-mark**
Depends on https://github.com/hubmapconsortium/vitessce/pull/585

Update: the above pull request has been merged and version `0.1.5` of Vitessce released.

This pull request adds an "expand" button on the right side of the "Visualization" heading. A snackbar/toast appears to tell the user that the Escape key can be used to minimize/collapse out of expanded mode.

### CODE REVIEW

File: context/app/static/js/components/Detail/Visualization/style.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  display: flex;
+  margin-left: auto;
+`;
+
+const FlexBottomRight = styled(FlexRight)`
+  align-items: flex-end;
+`;
+
+const StyledFooterText = styled.p`
+  line-height: 1.5;
```
Comment (john-conroy) [inline context/app/static/js/components/Detail/Visualization/style.js:49]:
```suggestion
  text-align: right;
  line-height: 1.5;
```

Comment (john-conroy) [reply context/app/static/js/components/Detail/Visualization/style.js:49]: I think just aligning the text and removing FlexContainer, FlexRight and FlexBottomRight in `style.js` and the component implementation file should work.

File: context/app/static/js/components/Detail/Visualization/style.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+const TopSnackbar = styled(Snackbar)`
+  top: ${headerFixedHeight + 10}px;
+`;
+
+const ExpandableDiv = styled.div`
+  top: ${(props) => (props.$isExpanded ? `${headerFixedHeight}px` : 'auto')};
+  left: ${(props) => (props.$isExpanded ? '0' : 'auto')};
+  position: ${(props) => (props.$isExpanded ? 'fixed' : 'relative')};
+  height: ${(props) => (props.$isExpanded ? `calc(100vh - ${headerFixedHeight}px)` : `${vitessceFixedHeight}px`)};
+  width: ${(props) => (props.$isExpanded ? '100%' : '100%')};
```
Comment (john-conroy) [inline context/app/static/js/components/Detail/Visualization/style.js:26]:
```suggestion
  width: 100%;
```

File: context/app/static/js/components/Detail/Visualization/style.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  margin-left: auto;
+`;
+
+const FlexBottomRight = styled(FlexRight)`
+  align-items: flex-end;
+`;
+
+const StyledFooterText = styled.p`
+  line-height: 1.5;
+  font-size: 0.75rem;
```
Comment (john-conroy) [inline context/app/static/js/components/Detail/Visualization/style.js:50]:
```suggestion
```

File: context/app/static/js/components/Detail/Visualization/style.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  margin: 0;
+
+  > a {
+    font-size: 0.75rem;
+  }
```
Comment (john-conroy) [inline context/app/static/js/components/Detail/Visualization/style.js:55]:
```suggestion
```

File: context/app/static/js/components/Detail/Visualization/style.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
+import styled from 'styled-components';
+import Button from '@material-ui/core/Button';
```
Comment (john-conroy) [inline context/app/static/js/components/Detail/Visualization/style.js:2]:
```suggestion
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';

```

File: context/app/static/js/components/Detail/Visualization/style.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+const FlexRight = styled.div`
+  display: flex;
+  margin-left: auto;
+`;
+
+const FlexBottomRight = styled(FlexRight)`
+  align-items: flex-end;
+`;
+
+const StyledFooterText = styled.p`
```
Comment (john-conroy) [inline context/app/static/js/components/Detail/Visualization/style.js:48]:
```suggestion
const StyledFooterText = styled(Typography)`
```

File: context/app/static/js/components/Detail/Visualization/Visualization.jsx [hunk 7]
```diff
@@ -1,22 +1,83 @@
...
+          />
+          <Vitessce config={vitData} theme="light" height={isExpanded ? null : vitessceFixedHeight} />
+        </ExpandableDiv>
       </Paper>
-    </SectionContainer>
+      <FlexContainer>
+        <FlexBottomRight>
+          <StyledFooterText>
+            Powered by&nbsp;
+            <Link href="http://vitessce.io" variant="body1" target="_blank">
```
Comment (john-conroy) [inline context/app/static/js/components/Detail/Visualization/Visualization.jsx:74]:
```suggestion
          <StyledFooterText variant="body2">
            Powered by&nbsp;
            <Link href="http://vitessce.io" target="_blank">
```

Comment (john-conroy) [reply context/app/static/js/components/Detail/Visualization/Visualization.jsx:74]: Slightly larger (0.80rem vs 0.75rem), but good to take typography sizing from the theme.

Comment (john-conroy) [reply context/app/static/js/components/Detail/Visualization/Visualization.jsx:74]: Also do we want to add `rel="noopener noreferrer"` to the Link?

File: context/app/static/js/components/Detail/Visualization/Visualization.jsx [hunk 8]
```diff
@@ -1,22 +1,83 @@
...
+
+  function handleExpand() {
+    setIsExpanded(true);
+    setIsSnackbarOpen(true);
+  }
+
+  function handleCollapse() {
+    setIsExpanded(false);
+    setIsSnackbarOpen(false);
+  }
```
Comment (john-conroy) [inline context/app/static/js/components/Detail/Visualization/Visualization.jsx:33]: Do we need both `setIsExpanded` and `setIsSnackBarOpen` if their values are in-sync?

File: context/app/static/js/components/Detail/Visualization/style.js (hunk 1 above)
Comment (john-conroy) [reply context/app/static/js/components/Detail/Visualization/style.js:49]: I should probably take another look at the Summary component I created to see if I can implement similar changes.

File: context/app/static/js/components/Detail/Visualization/Visualization.jsx (hunk 8 above)
Comment (keller-mark) [reply context/app/static/js/components/Detail/Visualization/Visualization.jsx:33]: Their values are not always in sync, the Snackbar component sets isSnackbarOpen to `false` after 4 seconds when it disappears.

**Comment by keller-mark** [comment]
The expanded width issue should be fixed by the vitessce -> and vitessce-grid pull request dependencies. I am just waiting for those to be reviewed.

**Comment by keller-mark** [comment]
Thank you for all of your suggestions @john-conroy!

File: context/package.json [hunk 9]
```diff
@@ -62,6 +62,7 @@
     "dev-server": "webpack-dev-server --config build-utils/webpack.dev.js --hot",
     "test": "jest",
     "test:watch": "jest --watch",
-    "test:coverage": "jest --coverage"
+    "test:coverage": "jest --coverage",
+    "lint-fix": "eslint -c .eslintrc.yml --ext .js --fix app/static/js/"
```
Comment (mccalluc) [inline context/package.json:66]: - I think we're also using `.jsx` extensions?
- Mention in the readme?

Comment (keller-mark) [reply context/package.json]: Oh right, I have updated this.

**Comment by keller-mark** [comment]
When Vitessce is expanded the scrollbar still appears. A common way to solve this issue is to add `overflow: hidden` to the `<body>` element. This is what material-ui and bootstrap do when they render their Modal components https://material-ui.com/components/modal/ https://getbootstrap.com/docs/4.5/components/modal/#live-demo

Is everyone OK with this approach? Note that `body` is outside the React scope.

File: context/app/static/js/components/Detail/Visualization/style.js [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+    display: block;
+    height: ${(props) => (props.$isExpanded ? `calc(100vh - ${headerFixedHeight}px)` : 'auto')};
+    width: 100%;
+  }
+`;
+
+const StyledFooterText = styled(Typography)`
+  line-height: 1.5;
+  text-align: right;
+  width: 100%;
```
Comment (john-conroy) [inline context/app/static/js/components/Detail/Visualization/style.js:40]: Did you determine `width: 100%` was needed? Typography variant body2 is a p which is block-level.

**Comment by john-conroy** [comment]
> When Vitessce is expanded the scrollbar still appears. A common way to solve this issue is to add `overflow: hidden` to the `<body>` element. This is what material-ui and bootstrap do when they render their Modal components https://material-ui.com/components/modal/ https://getbootstrap.com/docs/4.5/components/modal/#live-demo
> 
> Is everyone OK with this approach? Note that `body` is outside the React scope.

Adding `overflow: hidden` to the `<body>` element only when Vitessce is expanded or some similar logic? I think this should be fine.

File: context/app/static/js/components/Detail/Visualization/style.js (hunk 10 above)
Comment (keller-mark) [reply context/app/static/js/components/Detail/Visualization/style.js]: Thanks for pointing this out, I have removed this rule.

**Pull request merged**